source/Plugins/InstrumentationRuntime/MainThreadChecker/MainThreadCheckerRuntime.cpp

```diff
@@ -109,12 +109,32 @@ static std::string TranslateObjCNameToSwiftName(std::string className,
                            swift::DeclVisibilityKind Reason) {
       if (result)
         return; // Take the first result.
-      if (swift::ClassDecl *cls = clang::dyn_cast<swift::ClassDecl>(VD)) {
-        auto funcs = cls->lookupDirect(selectorToLookup, true);
-        if (funcs.size() > 0) {
-          result = funcs.front()->getFullName();
+      swift::ClassDecl *cls = clang::dyn_cast<swift::ClassDecl>(VD);
+      if (!cls)
+        return;
+      auto funcs = cls->lookupDirect(selectorToLookup, true);
+      if (funcs.size() == 0)
+        return;
+
+      // If the decl is actually an accessor, use the property name instead.
+      swift::AbstractFunctionDecl *decl = funcs.front();
+      if (auto func = clang::dyn_cast<swift::FuncDecl>(decl)) {
+        swift::DeclContext *funcCtx = func->getParent();
+        // We need to loadAllMembers(), otherwise 'isAccessor' returns false.
+        if (auto extension = clang::dyn_cast<swift::ExtensionDecl>(funcCtx)) {
+          extension->loadAllMembers();
+        }
+        if (auto nominal = clang::dyn_cast<swift::NominalTypeDecl>(funcCtx)) {
+          nominal->loadAllMembers();
+        }
```

Comment:
It looks like ExtensionDecl and NominalTypeDecl are mutually non-inheriting, so it should be fine to use an 'else if' here?